pallets/whitelist/src/lib.rs

```diff
@@ -0,0 +1,181 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	pub use frame_support::{
+		pallet_prelude::*, traits::StorageVersion, transactional, PalletId, Parameter,
+	};
+	use frame_system::{
+		pallet_prelude::*,
+		{self as system},
+	};
+	use sp_std::prelude::*;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// Origin used to administer the pallet
+		type WhitelistManagerOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Relayer added to set
+		WhitelistAdded(T::AccountId),
+		/// Relayer removed from set
+		WhitelistRemoved(T::AccountId),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Whitelist already in set
+		WhitelistAlreadyExists,
+		/// Provided accountId is not a Whitelist
+		WhitelistInvalid,
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn whitelist_on)]
+	pub type WhitelistOn<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn whitelists)]
+	pub type Whitelists<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, bool, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Adds a new whitelist
+		#[pallet::call_index(0)]
+		#[pallet::weight(10_000)]
+		pub fn add_whitelist(origin: OriginFor<T>, v: T::AccountId) -> DispatchResult {
+			T::WhitelistManagerOrigin::ensure_origin(origin)?;
+			ensure!(!Self::is_whitelist(&v), Error::<T>::WhitelistAlreadyExists);
+			Whitelists::<T>::insert(&v, true);
+			Self::deposit_event(Event::WhitelistAdded(v));
+			Ok(())
+		}
+
+		/// Batch adding of new whitelists
+		#[pallet::call_index(1)]
+		#[pallet::weight(100_000)]
+		#[transactional]
+		pub fn batch_add_whitelists(origin: OriginFor<T>, vs: Vec<T::AccountId>) -> DispatchResult {
+			T::WhitelistManagerOrigin::ensure_origin(origin)?;
+			for v in vs {
+				ensure!(!Self::is_whitelist(&v), Error::<T>::WhitelistAlreadyExists);
```

Comment:
I'm a bit undecided, should we continue or break? Same for the removal.

Reply:
break. And I use [transactional] for making sure of it.